src/request/create/ChooseIdenticon.js

```diff
@@ -0,0 +1,128 @@
+
+        this.$generateMoreButton.addEventListener('click', this.generateIdenticons);
+        this.$backdrop.addEventListener('click', this._clearSelection);
+    }
+
+    /**
+     * @param {HTMLElement} [$el]
+     *
+     * @returns {HTMLElement}
+     */
+    static _createElement($el) {
+        $el = $el || document.createElement('div');
+
+        $el.innerHTML = `<h1 data-i18n="create-choose-identicon-header1">Choose Your Account Avatar</h1>
+            <h2>The Avatar will be 'unique' to this Account. You can not change it later.</h2>
+            <div class="grow"></div>
+            <div class="identicons">
+                <div class="loading center">
+                    <div class="loading-animation"></div>
+                    <h2>Mixing colors</h2>
+                </div>
+            </div>
+            <div class="grow"></div>
+            <a class="secondary generate-more center">Generate New</a>
+            <div class="grow"></div>
+            <div class="backdrop center">
+                <button>Confirm</button>
+                <a class="secondary">Back</a>
+            </div>`;
+
+        I18n.translateDom($el);
+
+        return $el;
+    }
+
+    /**
+     * @returns {HTMLElement}
+     */
+    getElement() {
+        return this.$el;
+    }
+
+    generateIdenticons() {
+        this.$el.classList.remove('active');
+
+        /** @type {{ [address: string]: Nimiq.KeyPair}} */
+        const keyPairs = {};
+
+        for (let i = 0; i < 7; i++) {
+            const keyPair = Nimiq.KeyPair.generate();
+            const address = keyPair.publicKey.toAddress().toUserFriendlyAddress();
+            keyPairs[address] = keyPair;
+        }
+
+        this._volatileKeys = keyPairs;
+
+        this.$identicons.textContent = '';
+
+        Object.keys(keyPairs).forEach(address => {
+            const identicon = new Identicon(address);
+            const $identicon = identicon.getElement();
+            this.$identicons.appendChild($identicon);
+            const $address = document.createElement('div');
+            $address.classList.add('address');
+            $address.textContent = address;
+            $identicon.appendChild($address);
+            $identicon.addEventListener('click', () => this._onIdenticonSelected($identicon, address));
+        });
+
+        setTimeout(() => this.$el.classList.add('active'), 100);
+    }
+
+    /**
+     * @param {HTMLElement} $identicon
+     * @param {string} address
+     * @private
+     */
+    _onIdenticonSelected($identicon, address) {
+        const $returningIdenticon = this.$el.querySelector('.identicon.returning');
+        if ($returningIdenticon) {
+            $returningIdenticon.classList.remove('returning');
+        }
+
+        this.$confirmButton.addEventListener('click', () => this.fire(
+            ChooseIdenticon.Events.CHOOSE_IDENTICON,
+            this._volatileKeys[address],
+        ));
```

Comment:
this click listener does not get revoked when the selected identicon is dismissed.

Reply:
You mean when generating new identicons?